components/AddDonation.js

```diff
@@ -0,0 +1,144 @@
+import styled from "styled-components";
+import { useState } from "react";
+import TextLengthCounter from "./Counter";
+
+export default function AddDonation({ createDonation }) {
+  const [isFormVisible, setIsFormVisible] = useState(false);
+  const [count, setCount] = useState(0);
+  const translationMap = {
+    fruitsVegetables: "Obst & Gemüse",
+    homemade: "Selbstgemacht",
+    pastry: "Backwaren",
+  };
+
+  function handleToggleForm() {
+    setIsFormVisible(!isFormVisible);
+  }
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const newDonationData = Object.fromEntries(formData);
+
+    createDonation(newDonationData);
+    setCount(0);
+    event.target.reset();
+    setIsFormVisible(!isFormVisible);
+  }
+
+  return (
+    <StyledDiv>
+      {!isFormVisible ? (
+        <StyledButton onClick={handleToggleForm}>
+          Neue Spende hinzufügen
+        </StyledButton>
+      ) : (
+        <StyledForm onSubmit={handleSubmit}>
+          <label htmlFor="category">Kategorie:</label>
+          <select name="category" id="category" required>
+            <option value="">-Wähle bitte eine Kategorie-</option>
+            <option value={translationMap.fruitsVegetables}>
+              Obst & Gemüse
+            </option>
+            <option value="homemade">Selbstgemacht</option>
+            <option value="pastry">Backwaren</option>
+            <option value="dairy">Käse, Eier & Molkerei</option>
+            <option value="frozen">Tiefkühlkost</option>
+            <option value="sweet&salty">Süßes & Salziges</option>
+            <option value="beverage">Getränke</option>
+            <option value="meat&fish">Fleisch & Fisch</option>
```

Comment:
You could use a dataset to dynamically play out the options with map(). This cleans up your code further and lets the options scale with it, should you ever want to provide another one.

Reply:
This is only useful if I want to dynamically add options. But this is not the case. i already discussed this with Coach Felix as well :)

Reply:
It's always useful to not hardcode stuff ;) I was not talking about storing options within a state, but using a dataset for it.

Reply:
changed it- can you please update  my request in slack 👀✅